packages/pigeon/lib/swift_generator.dart

```diff
@@ -188,30 +188,30 @@ import FlutterMacOS
             indent.write('if let ${field.name}List = $listValue as! [Any]? ');
             indent.addScoped('{', '}', () {
               indent.writeln(
-                  '${field.name} = $fieldType.fromList(${field.name}List as [Any])');
+                  '${field.name} = $fieldType.fromList(${field.name}List)');
             });
           } else if (!hostDatatype.isBuiltin &&
               customEnumNames.contains(field.type.baseName)) {
             indent.writeln('var ${field.name}: $fieldType? = nil');
-            indent.write('if let ${field.name}RawValue = $listValue as! Int? ');
+            indent.writeln(
+                'let enumVal$index: Int? = ${_castForceUnwrap(listValue, const TypeDeclaration(baseName: 'Int', isNullable: true), root)}');
+            indent.write('if let ${field.name}RawValue = enumVal$index ');
```

Comment:
Does swift allow explicit types in `if let` statements? I.e., could these lines collapse to:

```
if let ${field.name}RawValue : Int = ${_castForceUnwrap(...)} 
```

to avoid the intermediate variable?